hamp_radar/iquick.py

```diff
@@ -222,26 +399,49 @@ def read_iq(filename, postprocess=True):
 
 
 def decode_time(ds):
+    """
+    Replaces 'Tm' and 'time_milli' variables in dataset 'ds' with decoded time.
+
+    The function calculates the time by adding:
+    - A base time of "1970-01-01"
+    - 'Tm' multiplied by 1 second (expressed as 1,000,000,000 ns)
+    - 'time_milli' multiplied by 0.001 (expressed as ns)
+
+    The resulting time is then assigned to a new 'time' variable in the dataset.
+
+    Parameters:
+    ds (xarray.Dataset): The input dataset containing 'Tm' and 'time_milli'
+                         variables.
+
+    Returns:
+    xarray.Dataset: The dataset with 'Tm' and 'time_milli' variables replaced by
+                    new 'time' variable added.
+    """
     time = (
         np.datetime64("1970-01-01")
         + ds.Tm * np.timedelta64(1000000000, "ns")
-        + ds.time_milli * np.timedelta64(1000, "ns")
+        + ds.time_milli * np.timedelta64(1000, "ns") # [sic] 'time_milli' is microseconds 
     )
     return ds.drop_vars(["Tm", "time_milli"]).assign(time=time)
 
 
 def postprocess_iq(ds):
+    # TODO(ALL): move to new file
     return ds.pipe(decode_time)
 
 
 def main():
+    '''
+    Use an argument parser with positional argument for the filename to read IQ
+    data from the file called 'filename' and decode the time.
+    '''
```

Comment:
I'd assume every `main()` function would do something like using an argument parse. The real docs for this function can be seen when running the script on the command line with `--help`.

```suggestion
```